### PR TITLE
gh-2887 stopping of a Subgraph of type FederatedStore being added whi…

### LINKED ISSUE #2887: FederatedStore Double Cache Persistence bug
**Describe the bug**
If a Rest Service FederatedStore is started and a sub-FederatedStore is added. The sub store could be added with no persistent cache. This would cause the sub-FederatedStore to become unusable. due to being fetched from cache-built-used each time and retains no information.

If a Rest Service FederatedStore is started with No persistent cache, a sub-FederatedStore could be created with A persistent cache.

### PR DESCRIPTION
**Pull request by GCHQDev404**
…le Cache is of type non-persistent HashMapCache

# Related Issue

- Resolve #2887

### CODE REVIEW

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java [hunk 1]
```diff
@@ -93,6 +98,20 @@ public Void doOperation(final OP operation, final Context context, final Store s
...
 
+    private static void processNonViableFedStoreWithNonPersistentCaches(final GraphSerialisable graphSerialisable) throws OperationException {
+        final String storeClass = graphSerialisable.getStoreProperties().getStoreClass();
+        final Class<? extends Store> aClass;
+        try {
+            aClass = Class.forName(storeClass).asSubclass(Store.class);
+        } catch (ClassNotFoundException e) {
+            throw new GafferRuntimeException("Error creating Store class from properties", e);
+        }
+        if (FederatedStore.class.isAssignableFrom(aClass) && CacheServiceLoader.getService() instanceof HashMapCacheService) {
```
Comment (t92549) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java:109]: Maybe the various cache services could have a trait like `isPersistent` or something?

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java [hunk 2]
```diff
@@ -93,6 +98,20 @@ public Void doOperation(final OP operation, final Context context, final Store s
         return null;
     }
 
+    private static void processNonViableFedStoreWithNonPersistentCaches(final GraphSerialisable graphSerialisable) throws OperationException {
+        final String storeClass = graphSerialisable.getStoreProperties().getStoreClass();
+        final Class<? extends Store> aClass;
+        try {
+            aClass = Class.forName(storeClass).asSubclass(Store.class);
```
Comment (t92549) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java:105]: I think usually in Gaffer `clazz` is used but it's not a big deal

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java (hunk 1 above)
Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java:109]: Yes that was my thinking.

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java (hunk 2 above)
Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java:105]: it was autogenerated.

**Comment by GCHQDeveloper314** [comment]
This appears to be only a change to a test and nothing else, which seems to be incongruous with the original ticket. It looks like the ticket 2887 was actually fixed in the PR 2902 and not this one? Is this PR required or can the one file changes be added to that other PR?

**Comment by GCHQDev404** [comment]
> This appears to be only a change to a test and nothing else, which seems to be incongruous with the original ticket. It looks like the ticket 2887 was actually fixed in the PR 2902 and not this one? Is this PR required or can the one file changes be added to that other PR?

The ticket had a comment by me explaining, it. So a test was added to prove it.

**Review by GCHQDeveloper314 — DISMISSED** [review]

**Review by GCHQDeveloper314 — APPROVED** [review]

**Review by t92549 — APPROVED** [review]

**Review by t92549 — APPROVED** [review]

**Review by GCHQDeveloper314 — APPROVED** [review]

**Pull request merged**